fabric-extensibility-api-v1/src/testmod/java/net/fabricmc/fabric/test/extensibility/trident/TestTrident.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.test.extensibility.trident;
+
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.MovementType;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.projectile.PersistentProjectileEntity;
+import net.minecraft.entity.projectile.TridentEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.sound.SoundCategory;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.sound.SoundEvents;
+import net.minecraft.stat.Stats;
+import net.minecraft.util.Hand;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.TypedActionResult;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.extensibility.item.v1.trident.SimpleTridentItem;
+import net.fabricmc.fabric.api.extensibility.item.v1.trident.SimpleTridentItemEntity;
+
+/**
+ * This trident allows the user to riptide from lava and burns entities when thrown at them.
+ */
+public class TestTrident extends SimpleTridentItem {
+	public TestTrident(Item.Settings settings) {
+		super(settings, new Identifier("fabric-extensibility-api-v1-testmod", "textures/entity/test_trident.png"));
+	}
+
+	@Override
+	public TridentEntity getTridentEntity(TridentEntity trident) {
+		return new SimpleTridentItemEntity(trident) {
+			@Override
+			protected void onHit(LivingEntity target) {
+				super.onHit(target);
+				target.setOnFireFor(5);
+			}
+		};
+	}
+
+	@Override
+	public TypedActionResult<ItemStack> use(World world, PlayerEntity user, Hand hand) {
+		ItemStack itemStack = user.getStackInHand(hand);
+
+		if (itemStack.getDamage() >= itemStack.getMaxDamage() - 1) {
+			return TypedActionResult.fail(itemStack);
+		} else if (EnchantmentHelper.getRiptide(itemStack) > 0 && !(user.isTouchingWaterOrRain() || user.isInLava())) {
+			return TypedActionResult.fail(itemStack);
+		} else {
+			user.setCurrentHand(hand);
+			return TypedActionResult.consume(itemStack);
+		}
+	}
+
+	@Override
+	public void onStoppedUsing(ItemStack stack, World world, LivingEntity user, int remainingUseTicks) {
```

Comment:
This is not really a basic test mod?

Reply:
Ah ok. i was just testing out how to modify some features that exist in Minecraft code. i can remove them